Parquet/ResultListener/ParquetResultListener.cs

```diff
@@ -133,15 +129,24 @@ private ParquetFile GetOrCreateParquetFile(SchemaBuilder schema, string path)
                     Directory.CreateDirectory(dirPath);
                 }
 
-                file = new ParquetFile(schema, path);
+                file = new ParquetFile(builder.ToSchema(), path);
                 _parquetFiles[path] = file;
             }
-            else if (!file.CanContain(schema))
+            
+            if (!file.CanContain(builder.ToSchema()))
             {
+                builder.Union(file.Schema);
                 file.Dispose();
-                file = new ParquetFile(schema, path);
+                string tmpPath = path + ".tmp";
```

Comment:
What happens if you already have a lingering .tmp file?

Reply:
It should get overriden, which is what we want as the file has a .tmp tag we shouldn't keep it imo.